lumicks/pylake/nb_widgets/image_editor.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+from dataclasses import dataclass, field
+import matplotlib.pyplot as plt
+from matplotlib.axes import Axes
+
+
+class ImageStackAxes(Axes):
+    def __init__(
+        self, *args, image=None, frame=0, channel="rgb", show_title=True, plot_kwargs={}, **kwargs
+    ):
+        """Custom axes to handle mouse events for rotating images about a defined tether.
+
+        Parameters
+        ----------
+        *args
+            positional arguments, forwarded to superclass
+        image : lk.CorrelatedStack
+            image object
+        frame : int
+            initial frame index to plot
+        channel : {'rgb', 'red', 'green', 'blue', None}
+            Channel to plot for RGB images (None defaults to 'rgb')
+            Not used for grayscale images
+        show_title : bool
+            Controls display of auto-generated plot title
+        plot_kwargs : dict
+            plotting keyword arguments, forwarded to `Axes.imshow()`
+        **kwargs
+            keyword arguments, forwarded to superclass
+        """
+        super().__init__(*args, **kwargs)
+
+        # we use a list to store the reference to the image object so that, in the widget,
+        # we can grab a reference to this and keep the stored information even if the axes
+        # are closed
```

Comment:
agreed it's a bit weird. thinking about it more, I've decided to go with my initial idea of just holding a reference to the axes inside the widget. The downside of this is that the axes object isn't garbage collected until the widget falls out of scope (even if the figure is closed) but I don't think this will cause issues unless you're keeping lots of instances in memory. new version is a lot cleaner I think

Reply:
I agree that the new version is easier to read and harder to accidentally misuse.